paperqa/docs.py

```diff
@@ -317,6 +317,11 @@ async def aadd(  # noqa: PLR0912
             clean_text = result.text.strip("`")
             if clean_text.startswith("json"):
                 clean_text = clean_text.replace("json", "", 1)
+            # there should be no nesting here, so we can just split on curlies
+            # Note - this is why we cannot extract it to a general method
+            # because JSON in general has nested curlies
+            clean_text = clean_text.split("{", 1)[-1].split("}", 1)[0]
+            clean_text = "{" + clean_text + "}"
```

Comment:
It looks like in that linked issue, the model responded with JSON and also an explanation. The trailing explanation broke `json.loads`.

I would say your comment here does not seem to track with what is happening. The code is stripping the leading/trailing text around the JSON, but your comment talks about nesting of JSON

Can you update the comment to say that its just stripping leading/trailing text?

Reply:
I made a longer docstring explaining what clean_text is trying to do.

My comment was specifically about the fact that the code I wrote will not work on JSON in general, anticipating that the code looks incorrect. I wrote a longer comment about the rationale and goals of the code

Reply:
Yeah nice work on the comment, it makes sense 👍